lib/assetreqs/assetreqs.go

```diff
@@ -55,7 +67,53 @@ func New(token string, assetName string, orderNum string, cadenceName string, ca
 		fPath: filePath,
 		fName: fileName,
 		oFmt:  outputFormat,
+	}, nil
+}
+
+func ordNumIsValidFmt(orderNum string) (v bool, err error) {
+	// Note: because of no look-ahead operator (?) in go regexp, this takes multiple evaluations.
+	// Make sure that the given order number has at least one number and at least one letter, contains only digits and
+	// letters, and is 6 characters long.
+
+	// Test length = 6.
+	if len(orderNum) != 6 {
+		return false, nil
+	}
+
+	// Length is correct - make sure it doesn't contain something other than letters or digits.
+	invalid, err := hasNonLetterOrNonDigit(orderNum)
+	if err != nil {
+		return false, err
+	}
+	if invalid {
+		// contains something other than letters and digits - invalid
+		return false, err
 	}
+
+	// Make sure it contains at least one letter.
+	v, err = hasLetter(orderNum)
+	if err != nil {
+		return false, err
+	}
+	if !v {
+		// doesn't contain a letter
+		return false, err
+	}
+
+	// Contains at least one letter - make sure it contains at least one digit.
+	return hasDigit(orderNum)
+}
+
+func hasLetter(str string) (bool, error) {
+	return regexp.Match("[a-zA-Z]+", []byte(str))
+}
+
+func hasDigit(str string) (bool, error) {
```

Comment:
May not be needed based on other comments.

Reply:
the point of this fix is to be able to catch an argument accident before passing the call off to the API (see linked issue https://github.com/sassoftware/viya4-orders-cli/issues/7). now - that said, since this was originally reported, the behavior has changed - instead of a 500, the example in the issue description now returns 404 / order number not found, so it's a little more user friendly and we could choose to NOFIX this if we decide that's acceptable - do you think it's acceptable? i certainly wouldn't call it "unhandled".

Reply:
Yeah that makes sense to me!